### PR TITLE
Broker Updates

### PR DESCRIPTION
**Pull request by trwalke**
This PR Enables broker, fixes an exception issue and reapplies an older PR that was lost due to the branch being deleted.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1591

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs [hunk 1]
```diff
@@ -58,12 +52,10 @@ public async Task<MsalTokenResponse> AcquireTokenUsingBrokerAsync(Dictionary<str
...
             _androidBrokerTokenResponse = null;
             _correlationId = AndroidBrokerHelper.GetValueFromBrokerPayload(brokerPayload, BrokerParameter.CorrelationId);
-            //Need to disable warning for non awaited async call.
+
             try
             {
-#pragma warning disable CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
-                new TaskFactory().StartNew(() => AcquireTokenInternalAsync(brokerPayload));
-#pragma warning restore CS4014 // Because this call is not awaited, execution of the current method continues before the call is completed
+                await Task.Run(() => AcquireTokenInternalAsync(brokerPayload)).ConfigureAwait(false);
```
Comment (trwalke) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs]: Exceptions were getting lost when I used TaskFactory().StartNew() so i had to revert
@bgavrilMS

Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs]: Why not just `await AcquireTokenInteranalAsync` ?

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs [hunk 2]
```diff
@@ -43,7 +38,6 @@ public AndroidBroker(CoreUIParent uiParent, ICoreLogger logger)
             _activity = uiParent?.Activity;
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
             _brokerHelper = new AndroidBrokerHelper(Application.Context, logger);
-            _readyForResponse = new SemaphoreSlim(0);
         }
 
         public bool CanInvokeBroker()
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs]: Can you test what happens if you invoke the broker twice? I.e. call AcquireTokenInteractive and then call it imediatly again, without completing the broker flow. This should be a test for both iOS and Android

File: src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs [hunk 3]
```diff
@@ -74,11 +66,10 @@ public async Task<MsalTokenResponse> AcquireTokenUsingBrokerAsync(Dictionary<str
                     throw new MsalException(MsalError.AndroidBrokerOperationFailed, ex.Message, ex);
             }
 
-            await _readyForResponse.WaitAsync().ConfigureAwait(false);
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs]: Could you test what happens now if the broker is not installed but users configures WithBroker(true) ?

**Comment by trwalke** [comment]
Manual testing resulted in errors so I will need to revert the async change and add the semephore back in. I spoke with henrik about this and he agrees that it is the best way to go for now. I am moving these changes here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1610 because jenny has some changes that she made to broker code to fix a bug.

**Pull request closed without merging**